src/components/Dashboard/RewardsModule.js

```diff
@@ -0,0 +1,346 @@
+import React from 'react'
+import {
+  Box,
+  Button,
+  GU,
+  Help,
+  Info,
+  textStyle,
+  useTheme,
+  Link,
+} from '@aragon/ui'
+
+import Loading from './Loading'
+import NoRewards from './NoRewards'
+
+import { useCourtConfig } from '../../providers/CourtConfig'
+import { useConnectedAccount } from '../../providers/Web3'
+
+import { formatTokenAmount, bigNum } from '../../lib/math-utils'
+
+const useTotalDisputesFees = (arbitrableFees, appealFees) => {
+  const totalArbitrableFees = arbitrableFees
+    ? arbitrableFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+    : bigNum(0)
+
+  const totalAppealFees = appealFees
+    ? appealFees.reduce((acc, fee) => acc.add(fee.amount), bigNum(0))
+    : bigNum(0)
+
+  return { totalArbitrableFees, totalAppealFees }
+}
+
+// Ruling Fees => ANJ fees
+// Dispute fees + Appeal Fees => DAI fees
+const RewardsModule = React.memo(function RewardsModule({
+  rewards,
+  loading,
+  onSettleReward,
+  onSettleAppealDeposit,
+}) {
+  const connectedAccount = useConnectedAccount()
+
+  // Form submission
+  const handleFormSubmit = async event => {
+    event.preventDefault()
+
+    if (!rewards) return
+
+    try {
+      // Claim all arbitrable fee rewards
+      const arbitrableTxs = []
+      for (const arbitrableFee of rewards.arbitrableFees) {
+        const { disputeId, rounds } = arbitrableFee
+        for (const roundId of rounds) {
+          arbitrableTxs.push(
+            await onSettleReward(disputeId, roundId, connectedAccount)
+          )
+        }
+      }
+
+      await Promise.all(arbitrableTxs.map(tx => tx.wait()))
+
+      // Claim all appeal fee rewards
+      const appealTxs = []
+      for (const appealFee of rewards.appealFees) {
+        const { disputeId, rounds } = appealFee
+        for (const roundId of rounds) {
+          appealTxs.push(await onSettleAppealDeposit(disputeId, roundId))
+        }
+      }
+
+      await Promise.all(appealTxs.map(tx => tx.wait()))
```

Comment:
Wondering if it would make sense to push both the `arbitrableTxs` and `appealTxs` into a single array and then do `Promise.all()`.

Right now if you had both, you would have to wait for the first set to get mined before the next set would appear, right?

Reply:
Yes totally makes sense, will add this change along the one above in the next PR which includes the additional tx from claiming the fees from the treasury